extensions/reactive-messaging-http/runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/HttpMessage.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.reactivemessaging.http.runtime;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.Metadata;
+
+/**
+ * used by http source
+ * 
+ * @param <T> payload type
+ */
+class HttpMessage<T> implements Message<T> {
+
+    private final T payload;
+    private final Runnable successHandler;
+    private final Consumer<Throwable> failureHandler;
+    private final Metadata metadata;
+
+    HttpMessage(T payload, IncomingHttpMetadata requestMetadata, Runnable successHandler,
+            Consumer<Throwable> failureHandler) {
+        this.payload = payload;
+        this.successHandler = successHandler;
+        this.failureHandler = failureHandler;
+        metadata = Metadata.of(requestMetadata);
```

Comment:
@kenfinnigan Should we do something around Open Telemetry tracing here?

Reply:
Ideally yes, but we don't have OpenTelemetry in Quarkus yet. It will need to wait